### PR TITLE
Respect MLFLow experiment environment variable

### PR DESCRIPTION
**Pull request by aspfohl**
This removes the default of naming all MLFLow experiments `my-mlflow-experiment`, and respects the environment variable MLFLOW_EXPERIMENT_NAME https://mlflow.org/docs/latest/python_api/mlflow.html?highlight=mlflow_experiment_name#mlflow.start_run

Slightly updates the docs about default behavior for runs and experiments

### CODE REVIEW

File: composer/loggers/mlflow_logger.py [hunk 1]
```diff
@@ -62,7 +67,8 @@ def init(self, state: State, logger: Logger) -> None:
             self.run_name += f'-rank{dist.get_global_rank()}'
 
         if self.experiment_name is None:
-            self.experiment_name = 'my-mlflow-experiment'
+            env_var = mlflow.environment_variables.MLFLOW_EXPERIMENT_NAME.name
```
Comment (eracah) [inline composer/loggers/mlflow_logger.py:70]: can we also check for `MLFLOW_EXPERIMENT_ID` as well here?

File: composer/loggers/mlflow_logger.py [hunk 2]
```diff
@@ -62,7 +67,8 @@ def init(self, state: State, logger: Logger) -> None:
             self.run_name += f'-rank{dist.get_global_rank()}'
```
Comment (eracah) [inline composer/loggers/mlflow_logger.py:67]: Can we check for the `MLFLOW_RUN_ID` [env variable](https://mlflow.org/docs/latest/python_api/mlflow.environment_variables.html?highlight=environment%20variables#mlflow.environment_variables.MLFLOW_RUN_ID) as well? Should a similar change to line 63

**Review by eracah — COMMENTED** [review]
Good catch. Thanks for doing this, @aspfohl ! Can you also add some unit tests to https://github.com/mosaicml/composer/blob/dev/tests/loggers/test_mlflow_logger.py#L43-L48

**Review by Skylion007 — APPROVED** [review]

**Review by eracah — APPROVED** [review]
Nice LGTM

**Pull request merged**